### PR TITLE
Add Milestone 2 documentation.

### PR DESCRIPTION
**Pull request by calebbrown**
Milestone 2 doc covering the (almost finished) production implementation.

### CODE REVIEW

File: docs/design/milestone_2.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+## Design
+
+### Enumeration
+
+![GitHub Enumeration Infra Design](images/github-enumeration-infra.svg)
+
+#### Initiation
+
+A Kubernetes schedule starts the enumeration process.
```
Comment (oliverchang) [inline docs/design/milestone_2.md:82]: Should we link to the kubernetes config?

File: docs/design/milestone_2.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+[Scalable Scorecard](https://github.com/ossf/scorecard/blob/main/docs/design/scalable_scorecards.md).
+
+Productionizing the collection of signals for calculating a criticality score
+aligns well with this infrastructure.
+
+## Design
+
+### Enumeration
+
+![GitHub Enumeration Infra Design](images/github-enumeration-infra.svg)
```
Comment (oliverchang) [inline docs/design/milestone_2.md:78]: Maybe link to the code for the entry point ?

File: docs/design/milestone_2.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    - writing the shards to storage
+- Data-transfer library ([ref](https://github.com/ossf/scorecard/blob/main/cron/internal/bq/main.go))
+  - A reusable library for determining if all shards are present, and: iterating
+    over the files (e.g. BigQuery loader) and/or iterating over each record in
+    the resulting CSV files.
+
+Additionally, the following changes need to be made to the Criticality Score
+project:
+
+- Implement a criticality score specific worker
```
Comment (oliverchang) [inline docs/design/milestone_2.md:183]: nit: link to the final implementation of the worker?

File: docs/design/milestone_2.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+    - use the "worker library" above to consume shards of work from a PubSub
+      channel.
+    - write each shard as a JSON file to a primary bucket location for BigQuery
+      import.
+    - write each shard also as a CSV file to a secondary bucket location for
+      aggregation into a large CSV file.
+    - consume configuration information from the scorecard config.yaml file,
+      and environment variables.
+    - apply a default score during collection.
+- Implement a criticality score specific data-transfer service to combine CSV
```
Comment (oliverchang) [inline docs/design/milestone_2.md:196]: nit: link to code?

**Review by oliverchang — APPROVED** [review]
thanks! This looks great! 

Just some minor comments:
- The SVG has a transparent background which makes the arrows difficult to see in dark mode.
- Can we have more links to the code that was implemented? This would be very helpful for any other contributors to follow along. 
-

File: docs/design/milestone_2.md (hunk 4 above)
Comment (calebbrown) [reply docs/design/milestone_2.md:196]: Done

File: docs/design/milestone_2.md (hunk 3 above)
Comment (calebbrown) [reply docs/design/milestone_2.md]: Done

File: docs/design/milestone_2.md (hunk 2 above)
Comment (calebbrown) [reply docs/design/milestone_2.md]: Done

File: docs/design/milestone_2.md (hunk 1 above)
Comment (calebbrown) [reply docs/design/milestone_2.md]: Done

File: docs/design/milestone_2.md [hunk 5]
```diff
@@ -256,5 +259,4 @@ Phase 2: staging -> production
   - On commit - staging will be updated to the latest images.
   - Staging will use a limited set of data (O(1000) repos) to allow for regular
     runs.
-  - Promtion from staging to production will be a manual step, likely involving
-    a manual release in the GitHub repo.
+  - Promtion from staging to production will be a manual step.
```
Comment (oliverchang) [inline docs/design/milestone_2.md:262]: nit: fix spelling of Promotion.

**Review by oliverchang — DISMISSED** [review]

**Review by oliverchang — APPROVED** [review]

**Pull request merged**